sw/device/tests/dif/dif_keymgr_unittest.cc

```diff
@@ -0,0 +1,1114 @@
+                    .value = 1,
+                }});
+  EXPECT_READ32(GetMaxKeyVersionRegInfo(GetParam()).wen_offset, 0);
+
+  EXPECT_EQ(dif_keymgr_advance_state(&keymgr_, &kStateParams),
+            kDifKeymgrLockableLocked);
+}
+
+TEST_P(AdvanceToOperational, Success) {
+  ExpectIdleAtState(GetParam());
+  EXPECT_READ32(KEYMGR_SW_BINDING_REGWEN_REG_OFFSET,
+                {{
+                    .offset = KEYMGR_SW_BINDING_REGWEN_EN_BIT,
+                    .value = 1,
+                }});
+  auto reg_info = GetMaxKeyVersionRegInfo(GetParam());
+  EXPECT_READ32(reg_info.wen_offset, {{
+                                         .offset = reg_info.wen_bit_index,
+                                         .value = 1,
+                                     }});
+  size_t binding_len = sizeof(kStateParams.binding_value) /
+                       sizeof(kStateParams.binding_value[0]);
+  for (size_t i = 0; i < binding_len; ++i) {
+    EXPECT_WRITE32(KEYMGR_SW_BINDING_0_REG_OFFSET + i * 4,
+                   kStateParams.binding_value[i]);
+  }
+  EXPECT_WRITE32(KEYMGR_SW_BINDING_REGWEN_REG_OFFSET, 0);
+  EXPECT_WRITE32(reg_info.offset, kStateParams.max_key_version);
+  EXPECT_WRITE32(reg_info.wen_offset, 0);
+  ExpectOperationStart({
+      .dest_sel = KEYMGR_CONTROL_DEST_SEL_VALUE_NONE,
+      .operation = KEYMGR_CONTROL_OPERATION_VALUE_ADVANCE,
+  });
+
+  EXPECT_EQ(dif_keymgr_advance_state(&keymgr_, &kStateParams),
+            kDifKeymgrLockableOk);
+}
+
+TEST_P(AdvanceToNonOperational, Success) {
+  ExpectIdleAtState(GetParam());
+  ExpectOperationStart({
+      .dest_sel = KEYMGR_CONTROL_DEST_SEL_VALUE_NONE,
+      .operation = KEYMGR_CONTROL_OPERATION_VALUE_ADVANCE,
+  });
+
+  EXPECT_EQ(dif_keymgr_advance_state(&keymgr_, nullptr), kDifKeymgrLockableOk);
+}
+
+class DisableTest : public DifKeymgrInitialized {};
+
+TEST_F(DisableTest, BadArgs) {
+  EXPECT_EQ(dif_keymgr_disable(nullptr), kDifKeymgrLockableBadArg);
+}
+
+TEST_F(DisableTest, LockedBusy) {
+  ExpectBusy();
+  EXPECT_EQ(dif_keymgr_disable(&keymgr_), kDifKeymgrLockableLocked);
+}
+
+TEST_F(DisableTest, LockedConfig) {
+  ExpectLockedConfig();
+
+  EXPECT_EQ(dif_keymgr_disable(&keymgr_), kDifKeymgrLockableLocked);
+}
+
+TEST_F(DisableTest, Disable) {
+  ExpectIdle();
+  ExpectOperationStart({
+      .dest_sel = KEYMGR_CONTROL_DEST_SEL_VALUE_NONE,
+      .operation = KEYMGR_CONTROL_OPERATION_VALUE_DISABLE,
+  });
+
+  EXPECT_EQ(dif_keymgr_disable(&keymgr_), kDifKeymgrLockableOk);
+}
+
+TEST_P(BadArgsTwo, GetStatusCodes) {
+  auto keymgr = GetGoodBadPtrArg<dif_keymgr_t>(std::get<0>(GetParam()));
+  auto status_codes =
+      GetGoodBadPtrArg<dif_keymgr_status_codes_t>(std::get<1>(GetParam()));
+
+  EXPECT_EQ(dif_keymgr_get_status_codes(keymgr, status_codes),
+            kDifKeymgrBadArg);
+}
+
+struct GetStatusCodesTestCase {
+  /**
+   * Values of OP_STATUS or ERR_CODE registers.
+   */
+  std::initializer_list<mock_mmio::BitField> reg_val;
```

Comment:
I'd be careful here. The lifetime of std::initializer_list isn't something that C++ is going to do you any favors with. Can you std::vector this?

Reply:
Done, had to make a small change in `mock_mmio.h`.